source/material_model/visco_plastic.cc

```diff
@@ -297,6 +297,27 @@ namespace aspect
                   elastic_out->elastic_shear_moduli[i] = average_elastic_shear_moduli[i];
                 }
             }
+
+          // Prescribed dilation
+          double average_friction_angle = 0.;
+          double dilation_rate = 0.;
+          if (this->get_parameters().enable_prescribed_dilation)
+            {
+              MaterialModel::PrescribedPlasticDilation<dim>
+              *prescribed_dilation = out.template get_additional_output<MaterialModel::PrescribedPlasticDilation<dim>>();
+
+              if (prescribed_dilation)
+                {
+                  average_friction_angle = MaterialUtilities::average_value(volume_fractions,
+                                                                            isostrain_viscosities.current_friction_angles,
+                                                                            rheology->viscosity_averaging);
+
+                  if (plastic_yielding)
+                    dilation_rate = 2. * std::sin(average_friction_angle) * isostrain_viscosities.strain_rate_invariant;
+
+                  prescribed_dilation->dilation[i] = dilation_rate;
```

Comment:
Why don't you assign it directly here?

Reply:
Also, do you need to set it =0; otherwise?

Reply:
> Why don't you assign it directly here?

The default value is set to 0 on line 303, and it is not assigned directly here as I was not sure how the logic for plastic_yielding = true or false would be worked in on a single line. Open to suggestions.